controllers/movies.ts

```diff
@@ -0,0 +1,302 @@
+import { Movie } from "../model/Movie";
+import { shortenURLs } from "../shorten-url";
+import { MovieParams, MovieQuerableParams } from "../util/types";
+
+class Movies {
+  async getAllMovies(params: MovieQuerableParams) {
+    const { actor, imdbSort, yearSort } = params;
+    try {
+      let query = Movie.createQueryBuilder("movie");
+      if (actor) {
+        query = query.where("movie.Actors LIKE :actor", {
+          actor: `%${actor}%`,
+        });
+      }
+      if (imdbSort) {
+        const imdbSortDirection =
+          imdbSort.toUpperCase() === "ASC" ? "ASC" : "DESC";
+        query = query.orderBy("movie.imdbRating", imdbSortDirection);
+      }
+
+      if (yearSort) {
+        const yearSortDirection =
+          yearSort.toUpperCase() === "ASC" ? "ASC" : "DESC";
+        query = query.orderBy("movie.Year", yearSortDirection);
+      }
+      let movies;
+
+      if (imdbSort) {
+        movies = await query
+          .select(["movie.Title", "movie.imdbRating"])
+          .getMany();
+      } else {
+        movies = await query.getMany();
+      }
+      return movies;
+    } catch (err: any) {
+      throw new Error("Movies not found");
+    }
+  }
+
+  async getMovieByImdID(imdb: any) {
+    try {
+      const movie = await Movie.findOne({
+        where: {
+          imdbID: imdb,
+        },
+      });
+      return movie;
+    } catch (err: any) {
+      throw new Error("Movie not found");
+    }
+  }
+
+  async deleteMovie(id: any) {
+    try {
+      const movie = (await Movie.findOne({
+        where: {
+          Id: id,
+        },
+      })) as Movie;
+      await Movie.remove(movie);
+      return movie;
```

Comment:
You shouldn't return the movie in the response after it is deleted

Reply:
Fixed, method deleteMovie now returns deletedMovieId.